WordPress/src/test/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsSettingsHelperTest.kt

```diff
@@ -0,0 +1,294 @@
+import org.wordpress.android.ui.prefs.AppPrefsWrapper
+import org.wordpress.android.util.config.BloggingPromptsEnhancementsFeatureConfig
+import org.wordpress.android.util.config.BloggingPromptsFeatureConfig
+import java.util.Date
+
+@ExperimentalCoroutinesApi
+class BloggingPromptsSettingsHelperTest : BaseUnitTest() {
+    @Mock
+    lateinit var bloggingRemindersStore: BloggingRemindersStore
+
+    @Mock
+    lateinit var selectedSiteRepository: SelectedSiteRepository
+
+    @Mock
+    lateinit var appPrefsWrapper: AppPrefsWrapper
+
+    @Mock
+    lateinit var bloggingPromptsFeatureConfig: BloggingPromptsFeatureConfig
+
+    @Mock
+    lateinit var bloggingPromptsEnhancementsFeatureConfig: BloggingPromptsEnhancementsFeatureConfig
+
+    lateinit var helper: BloggingPromptsSettingsHelper
+
+    @Before
+    fun setUp() {
+        helper = BloggingPromptsSettingsHelper(
+            bloggingRemindersStore,
+            selectedSiteRepository,
+            appPrefsWrapper,
+            bloggingPromptsFeatureConfig,
+            bloggingPromptsEnhancementsFeatureConfig,
+        )
+    }
+
+    @Test
+    fun `when getPromptsCardEnabledLiveData, then returns the store model isPromptsCardEnabled value`() {
+        val expectedState = true
+        val model = createRemindersModel(isPromptsCardEnabled = expectedState)
+        whenever(bloggingRemindersStore.bloggingRemindersModel(any())).doAnswer {
+            flowOf(model)
+        }
+
+        var result: Boolean? = null
+        helper.getPromptsCardEnabledLiveData(123).observeForever { result = it }
+
+        assertThat(result).isEqualTo(expectedState)
+    }
+
+    @Test
+    fun `when updatePromptsCardEnabledBlocking, then calls suspending version internally`() {
+        val spyHelper = spy(helper)
+
+        whenever(bloggingRemindersStore.bloggingRemindersModel(any())).doAnswer {
+            flowOf()
+        }
+
+        spyHelper.updatePromptsCardEnabledBlocking(123, isEnabled = true)
+
+        verifyBlocking(spyHelper) {
+            updatePromptsCardEnabled(any(), any())
+        }
+    }
+
+    @Test
+    fun `when updatePromptsCardEnabled, then updates the store model`() = test {
+        val expectedState = true
+        val model = createRemindersModel(isPromptsCardEnabled = false)
+        whenever(bloggingRemindersStore.bloggingRemindersModel(any())).doAnswer {
+            flowOf(model)
+        }
+
+        helper.updatePromptsCardEnabled(123, isEnabled = expectedState)
+
+        verify(bloggingRemindersStore).updateBloggingReminders(argThat { isPromptsCardEnabled == expectedState })
+    }
+
+    @Test
+    fun `given prompts FF is off and site is potential blog, when isPromptsFeatureAvailable, then returns false`() {
+        whenever(bloggingPromptsFeatureConfig.isEnabled()).thenReturn(false)
+        whenever(selectedSiteRepository.getSelectedSite()).thenReturn(
+            SiteModel().apply { setIsPotentialBloggingSite(true) }
```

Comment:
I was wondering if it makes sense to extract the code creating a `SiteModel` instance that repeats itself to a private method in this class 🤔